OSDOCS-11499: ROSA-HCP - Image registry config to customers (#82728)
openshift/openshift-docs: modules/images-configuration-parameters-hcp.adoc

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc

[id="images-configuration-parameters-hcp_{context}"]
= Image controller configuration parameters for {hcp-title}

> The `image.config.openshift.io/cluster` resource holds cluster-wide information about how to handle images. The resource exists, but it is read only and can only be changed through supported tools like ROSA CLI (`rosa`). The canonical, and only valid name is `cluster`. It can be configured in {product-rosa} {hcp} through ROSA CLI (`rosa`) commands.

Reviewer: ```suggestion
The `image.config.openshift.io/cluster` resource holds cluster-wide information about how to handle images. The resource exists, but it is read only and can only be changed through supported tools like ROSA CLI (`rosa`). The canonical and only valid name is `cluster`. It can be configured in {product-rosa} {hcp} through ROSA CLI (`rosa`) commands.
```

[NOTE]
====
Parameters such as `DisableScheduledImport`, `MaxImagesBulkImportedPerRepository`, `MaxScheduledImportsPerMinute`, `ScheduledImageImportMinimumIntervalSeconds`, `InternalRegistryHostname` are not configurable.
====

[cols="3a,8a",options="header"]
|===
|Parameters for ROSA CLI |Description

|`registry-config-allowed-registries`

> |Registries for which image pull and push actions are allowed. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. All other registries are blocked. The format should be a comma-separated list of allowed registries. For example: `allowed.io, allowed.io2`

Reviewer: Consistency with commas after the "For example"s, and period at the end:
```suggestion
|Registries for which image pull and push actions are allowed. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example, `reg1.io/myrepo/myapp:latest`. All other registries are blocked. The format should be a comma-separated list of allowed registries. For example, `allowed.io, allowed.io2`.
```

|`registry-config-insecure-registries`

> |Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. The format should be a comma-separated list of insecure registries. For example: `insecure.io, insecure.io2`

Reviewer: Same:
```suggestion
|Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example, `reg1.io/myrepo/myapp:latest`. The format should be a comma-separated list of insecure registries. For example, `insecure.io, insecure.io2`.
```

|`registry-config-blocked-registries`

> |Registries for which image pull and push actions are denied. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. All other registries are allowed. The format should be a comma-separated list of blocked registries. For example: `blocked.io, blocked.io2`

Reviewer: ```suggestion
|Registries for which image pull and push actions are denied. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example, `reg1.io/myrepo/myapp:latest`. All other registries are allowed. The format should be a comma-separated list of blocked registries. For example, `blocked.io, blocked.io2`.
```

|`registry-config-allowed-registries-for-import`

> |Contains configuration that determines how the container runtime should treat individual registries when accessing images for builds and pods. For example, whether or not to allow insecure access. It does not contain configuration for the internal cluster registry. Limits the container image registries from which normal users can import images. The format should be a comma-separated list of 'domainName:insecure'. 'domainName' specifies a domain name for the registry. 'insecure' indicates whether the registry is secure or insecure.

Reviewer: Backticks for markup:
```suggestion
|Contains configuration that determines how the container runtime should treat individual registries when accessing images for builds and pods. For example, whether or not to allow insecure access. It does not contain configuration for the internal cluster registry. Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure`. `domainName` specifies a domain name for the registry. `insecure` indicates whether the registry is secure or insecure.
```

|`registry-config-additional-trusted-ca`

> |A json file containing the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.

Reviewer: ```suggestion
|A JSON file containing the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.
```

|`registry-config-platform-allowlist`

> |A list of Red{nbsp}Hat registries is automatically allowed. This list can be periodically updated and impacted clusters will receive a notification with the new allowlist id. In such case, user needs to use this parameter to update to the expected id to the expected one.

Reviewer 2: Could you fix the wording here? Something like "from the previous expected id to the newly expected id?

Author: changed thanks

> |A list of Red{nbsp}Hat registries is automatically allowed. This list can be periodically updated and impacted clusters will receive a notification with the new allowlist id. In such case, user needs to use this parameter to update from the previous expected id to the newly expected id.

Reviewer: ```suggestion
|A list of Red{nbsp}Hat registries is automatically allowed. This list can be periodically updated and impacted clusters will receive a notification with the new allowlist ID. In such cases, the user must use this parameter to update from the previous expected ID to the newly expected ID.
```

|===

[WARNING]
====

> When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. For disconnected clusters, mirror registries should be added by the user itself.

Reviewer 3: ```suggestion
When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list is composed by `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running `rosa describe cluster`. 
```

Reviewer 4: >The current list is composed by

should become

> the current list consists of

Author: Changed in both places to "The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running `rosa describe cluster`."

> When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running `rosa describe cluster`.

Reviewer: ```suggestion
When the `allowedRegistries` parameter is defined, all registries are blocked unless explicitly listed. To prevent pod failure, a list of Red{nbsp}Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running the `rosa describe cluster` command.
```

|`registry-config-allowed-registries-for-import`
|Contains configuration that determines how the container runtime should treat individual registries when accessing images for builds and pods. For example, whether or not to allow insecure access. It does not contain configuration for the internal cluster registry.

|`registry-config-platform-allowlist`
|Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`.

|`registry-config-additional-trusted-ca`
| A json file containing the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.

|===

[WARNING]
====

> When the `allowedRegistries` parameter is defined, all registries, including the `registry.redhat.io` and `quay.io` registries and the default {product-registry}, are blocked unless explicitly listed. If you use the parameter, to prevent pod failure, add the `registry.redhat.io` and `quay.io` registries and the `internalRegistryHostname` to the `allowedRegistries` list, as they are required by payload images within your environment. For disconnected clusters, mirror registries should also be added.

Reviewer 3: The Red Hat list is automatically handled. I would change this to something like this.
```suggestion
When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. For disconnected clusters, mirror registries should be added by the user itself.
```

Author: updated

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc

[id="images-configuration-parameters-hcp_{context}"]
= Image controller configuration parameters for ROSA with HCP

The `image.config.openshift.io/cluster` resource holds cluster-wide information about how to handle images. The canonical, and only valid name is `cluster`. It can be configured in ROSA HCP through ROSA CLI commands.

[NOTE]
====
Parameters such as `DisableScheduledImport`, `MaxImagesBulkImportedPerRepository`, `MaxScheduledImportsPerMinute`, `ScheduledImageImportMinimumIntervalSeconds`, `InternalRegistryHostname` are not configurable.
====

[cols="3a,8a",options="header"]
|===
|Parameters for ROSA CLI |Description

|`registry-config-allowed-registries`

> |Registries for which image pull and push actions are allowed. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. All other registries are blocked.

Reviewer 5: I wonder if we should mention that it can be a comma-separated list of registries, meaning we can take multiple.

> |`registry-config-insecure-registries`

Reviewer 5: Same as insecure, I wonder if we should mention that it can be a comma-separated list of registries, meaning we can take multiple.

|Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`.

> The namespace for this config map is `openshift-config`. The format of the config map is to use the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.

Reviewer 5: @andreadecorte I feel like the namespace section should be removed, WDYT?

Reviewer 3: this paragraph needs to be removed, yes. (it is not for this parameter in any case)

Author: removed line 25

|`registry-config-blocked-registries`

> |Registries for which image pull and push actions are denied. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. All other registries are allowed.

Reviewer 5: I wonder if we should mention that it can be a comma-separated list of registries, meaning we can take multiple.

|`registry-config-allowed-registries-for-import`

> |Contains configuration that determines how the container runtime should treat individual registries when accessing images for builds and

Reviewer 5: Existing one already looks good, but it would be better to combine with "Limits the container image registries from which normal users can import images. The format should be a comma-separated list of 'domainName:insecure'. 'domainName' specifies a domain name for the registry. 'insecure' indicates whether the registry is secure or insecure." for a new paragraph.

pods. For example, whether or not to allow insecure access. It does not contain configuration for the internal cluster registry.

|`registry-config-platform-allowlist`

> |Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`.

Reviewer 5: For this one, @andreadecorte it's currently hidden. Do we plan to expose it in the doc?

Reviewer 3: My idea was to have this parameter not here, but in a dedicated section as it is a non functional parameter. Something like
Title: Platform allowlist
It is a list of platform mandatory registry.
It is visible in the rosa describe cluster.
It can be updated from time to time, and you will receive a Service Log with the new id. You can then update it by using rosa edit cluster --platform-allowlist-id <newid>. This will ensure you allow all the needed flows for the platform.

WDYT @chenz4027 ?

Reviewer 5: @andreadecorte I think it's valid. It's just that the deadline is next Tuesday, which is really tight. If we can prolong the deadline I think we can squeeze it in, otherwise it might be better if we modify the description instead of introducing a new section.

Reviewer 3: ok then quick proposal, at minimum we should:
1. move this as last parameter of the table
2. description
```
A list of Red Hat registries is automatically allowed. This list can be periodically updated and impacted clusters will receive a notification with the new allowlist id. In such case, user needs to use this parameter to update to the expected id to the expected one 
```

Author: moved and new description added

Author: moved the parameter to end and new description provided by @andreadecorte thanks

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc

[id="images-configuration-parameters-hcp_{context}"]
= Image controller configuration parameters for HCP

> The `image.config.openshift.io/cluster` resource holds cluster-wide information about how to handle images. The canonical, and only valid name is `cluster`. Its `spec` offers the following configuration parameters.

Reviewer 3: >Its `spec` offers the following configuration parameters.

I would change with

> It can be configured in ROSA HCP through ROSA CLI commands or something similar

Author: changed, thanks

[NOTE]
====

> Parameters such as `DisableScheduledImport`, `MaxImagesBulkImportedPerRepository`, `MaxScheduledImportsPerMinute`, `ScheduledImageImportMinimumIntervalSeconds`, `InternalRegistryHostname` are not configurable.

Reviewer 5: I don't think this part is applicable. So does line 8.

Author: @andreadecorte will I remove line 8 - 12:

"The `image.config.openshift.io/cluster` resource holds cluster-wide information about how to handle images. The canonical, and only valid name is `cluster`. It can be configured in ROSA HCP through ROSA CLI commands.

[NOTE]
====
Parameters such as `DisableScheduledImport`, `MaxImagesBulkImportedPerRepository`, `MaxScheduledImportsPerMinute`, `ScheduledImageImportMinimumIntervalSeconds`, `InternalRegistryHostname` are not configurable."

Author: added in following new content "
[NOTE]
====
 You can change any registry related parameter which will trigger a rollout across all machinepools (all machinepool nodes will be recreated, following pod draining from each node).
====

while on the meeting call with @chenz4027

|`registry-config-insecure-registries`
|Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`.

The namespace for this config map is `openshift-config`. The format of the config map is to use the registry hostname as the key, and the PEM-encoded certificate as the value, for each additional registry CA to trust.

|`registry-config-blocked-registries`
|Registries for which image pull and push actions are denied. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`. All other registries are allowed.

> |registry-config-allowed-registries-for-import`

Reviewer 5: Missing "`"  before registry

|Contains configuration that determines how the container runtime should treat individual registries when accessing images for builds and
pods. For instance, whether or not to allow insecure access. It does not contain configuration for the internal cluster registry.

`registry-config-platform-allowlist`: Registries which do not have a valid TLS certificate or only support HTTP connections. To specify all subdomains, add the asterisk (`\*`) wildcard character as a prefix to the domain name. For example, `*.example.com`. You can specify an individual repository within a registry. For example: `reg1.io/myrepo/myapp:latest`.

> 

Reviewer 5: Missing `registry-config-additional-trusted-ca` parameter.

Author: added, thanks

|===

[WARNING]
====

> When the `allowedRegistries` parameter is defined, all registries, including `registry.redhat.io` and `quay.io` registries and the default {product-registry}, are blocked unless explicitly listed. When using the parameter, to prevent pod failure, add all registries including the `registry.redhat.io` and `quay.io` registries and the `internalRegistryHostname` to the `allowedRegistries` list, as they are required by payload images within your environment. For disconnected clusters, mirror registries should also be added.

Reviewer 5: No internalRegistryHostname for hcp.

Author: @chenz4027 so remove the complete "Warning"?

====

The `status` field of the `image.config.openshift.io/cluster` resource holds observed values from the cluster.

[cols="3a,8a",options="header"]
|===
|Parameter |Description

> |`internalRegistryHostname`

Reviewer 5: Same here, could be removed.

Author: @chenz4027 remove complete table or just the "internalRgistryHOstname"

Author: sorry @chenz4027 next comment confirms whole table needs to be removed, thanks

|Set by the Image Registry Operator, which controls the `internalRegistryHostname`. It sets the hostname for the default {product-registry}. The value must be in `hostname[:port]` format. For backward compatibility, you can still use the `OPENSHIFT_DEFAULT_REGISTRY` environment variable, but this setting overrides the environment variable.

> |`externalRegistryHostnames`

Reviewer 5: This whole section can be removed as they are not applicable.

Author: removed complete table ,thanks
